activerecord/lib/active_record/migration.rb

```diff
@@ -575,6 +575,13 @@ def self.current_version
 
     MigrationFilenameRegexp = /\A([0-9]+)_([_a-z0-9]*)\.?([_a-z0-9]*)?\.rb\z/ # :nodoc:
 
+    def self.valid_version_format?(version_string)
+      [
+        MigrationFilenameRegexp,
+        /\A(?=\d)[_\d]+(?<=\d)\z/ # integer with optional underscores
```

Comment:
This regexp would match strings like `"1__2"`.  If we only want to match valid Ruby integer literals, I think the following would suffice:

```suggestion
        /\A\d(_?\d)*\z/ # integer with optional underscores
```

Reply:
Updated, and added test case.